### PR TITLE
ISPN-10770 Add generics to InvocationStage methods

### PR DESCRIPTION
**Pull request by wburns**
https://issues.jboss.org/browse/ISPN-10770

Typing allows for more developers using interceptor stack to more easily chain commands as they don't need to cast every time (which would lead to hopefully more non capturing lambdas).

I also fixed quite a few lambdas to be non capturing now, which should help performance a minor bit in some areas.

### CODE REVIEW

**Comment by wburns** [comment]
While there are quite a few changes, they are all rather trivial. Just changes or adds typing to the various methods.

Unfortunately due to how `QueueAsyncInvocationStage` works I was unable to add typing and have thus left it raw without generics.

**Comment by wburns** [comment]
Note that the changes to `PrefetchInterceptor` are required to fix a typing bug in Graal. It didn't like casting a `VisitableCommand` to a multi parameter type, where `VisitableCommand` was not the first type.

File: remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/ProtobufMetadataManagerInterceptor.java [hunk 1]
```diff
@@ -308,7 +307,7 @@ public Object visitPutMapCommand(InvocationContext ctx, PutMapCommand command) {
       invoker.running().invoke(ctx, cmd);
 
       return invokeNextThenAccept(ctx, command, (rCtx, rCommand, rv) -> {
-         long flagsBitSet = copyFlags(((PutMapCommand) rCommand));
+         long flagsBitSet = copyFlags((rCommand));
```
Comment (tristantarrant) [inline remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/ProtobufMetadataManagerInterceptor.java]: Double brackets ?

Comment (wburns) [reply remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/ProtobufMetadataManagerInterceptor.java]: Just missed removing them during my massive refactoring 👍

Comment (wburns) [reply remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/ProtobufMetadataManagerInterceptor.java]: Fixed.

**Comment by anistor** [comment]
LGTM also

**Comment by wburns** [comment]
Build should work now (darn imports)

**Comment by wburns** [comment]
Failures appear unrelated. I rebased to clear away other failures in case we need to run again.

**Comment by tristantarrant** [comment]
Merged, thanks

**Pull request merged**